secret.go

```diff
@@ -11,42 +17,93 @@ import (
 func applySecret(ctx *cli.Context) error {
 	baseURL := getNGBaseURL(ctx)
 	gitPat := ctx.String("token")
+	filePath := ctx.String("file")
 	gitPat = getGitSecret(gitPat)
+	secretType := ctx.String("secret-type")
+	requiresFile := isFileTypeSecret(secretType)
+	secretIdentifier := getSecretIdentifier(secretType)
+	secretName := getSecretName(secretType)
+	var secretBody HarnessSecret
+	var headers map[string]string
+	var err error
 
-	if gitPat == "" {
+	if requiresFile && filePath == "" {
+		println(fmt.Sprintf("Secret type %s requires file path to create a valid filetype secret", secretType))
+		return nil
+	}
+	if !requiresFile && gitPat == "" {
 		println("Secret cannot be an empty string")
 		return nil
 	}
-	createSecretURL := GetUrlWithQueryParams("", baseURL, "v2/secrets", map[string]string{
+	if requiresFile {
+		headers = make(map[string]string)
+		headers["Content-Type"] = "multipart/form-data"
+	}
+
+	createUrl := "v2/secrets"
```

Comment:
Let's define the `ENDPOINT` in the `constants.go` file as done for the other endpoints. For ex: `ENVIRONMENT_ENDPOINT`